### PR TITLE
 fix(lsp): treat empty string config value as None 

### LINKED ISSUE #14630: Treat `"deno.config": ""` as `"deno.config": null` in LSP
When editing the config in the VSCode settings UI, it is possible to set the setting to the empty string (which looks the same as having it set to `null` from the UI). The LSP should treat `null` and `""` the same.

### PR DESCRIPTION
**Pull request by crowlKats**
Fixes #14630

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
What about other field values that might contain empty strings? Could you update their handling too?

File: cli/lsp/config.rs [hunk 1]
```diff
@@ -720,4 +753,16 @@ mod tests {
...
   }
+
+  #[test]
+  fn test_empty_config() {
+    let mut config = Config::new();
+    config
+      .set_workspace_settings(json!({ "config": "" }))
+      .expect("could not update");
+    assert_eq!(
+      config.get_workspace_settings(),
```
Comment (bartlomieju) [inline cli/lsp/config.rs:764]: Could you add tests that check for empty strings in the values you changed?

Comment (crowlKats) [reply cli/lsp/config.rs:800]: done

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/lsp/config.rs [hunk 2]
```diff
@@ -266,8 +266,15 @@ fn default_to_true() -> bool {
   true
 }
 
+fn empty_string_none<'de, D: serde::Deserializer<'de>>(
+  d: D,
+) -> Result<Option<String>, D::Error> {
+  let o: Option<String> = Option::deserialize(d)?;
+  Ok(o.filter(|s| !s.is_empty()))
```
Comment (dsherret) [inline cli/lsp/config.rs:273]: I wonder if we should do `!s.trim().is_empty()`?

Comment (crowlKats) [reply cli/lsp/config.rs:273]: eh, seems a bit weird

**Pull request merged**